cluster/cluster.go

```diff
@@ -563,3 +571,101 @@ func MustGetSegmentConfiguration(connection *dbconn.DBConn, getMirrors ...bool)
 	gplog.FatalOnError(err)
 	return segConfigs
 }
+
+// GetSegmentConfigurationFromFile function to read segment configuration from dumpconfiguration file.
+// file should have the absolute path else if the file is empty it will generate default path coordinator_data_dir/gpsegconfig_dump
+// file should have the valid configuration format
+// dbid content role prole mode status port hostname address datadir
+func GetSegmentConfigurationFromFile(file string) ([]SegConfig, error) {
+	var conFilePath string
+	if file == "" {
+		filename := "gpsegconfig_dump" // filename which dump from fts process to master datadir
+		coordinator_datadir := os.Getenv("COORDINATOR_DATA_DIRECTORY")
+		if coordinator_datadir == "" {
+			coordinator_datadir = os.Getenv("MASTER_DATA_DIRECTORY")
+		}
+		if coordinator_datadir == "" {
+			errStr := "COORDINATOR_DATA_DIRECTORY is not set, not able to generate gpsegconfig_dump file path"
+			gplog.Error(errStr)
+			return nil, fmt.Errorf(errStr)
+		}
+		conFilePath = filepath.Join(coordinator_datadir, filename)
+	} else {
+		conFilePath = file
+	}
```

Comment:
I'm hesitant on adding adding this (COORDINATOR_DATA_DIRECTORY and MASTER_DATA_DIRECTORY) check for gp-common-go-libs.
Which ever program who wants to use the environment mentioned above,  should have already checked it; the library should handle just the filepath.

That's just my take, I'll let someone else chime in.

Reply:
Yeah I agree the calling function should just pass the entire filename to this function